Networking/NetworkingTests/Remote/PaymentsGatewayRemoteTests.swift

```diff
@@ -0,0 +1,42 @@
+import XCTest
+import TestKit
+
+@testable import Networking
+
+/// PaymentGatewayRemote Unit Tests
+///
+final class PaymentsGatewayRemoteTests: XCTestCase {
+
+    /// Dummy Network Wrapper
+    ///
+    let network = MockupNetwork()
+
+    /// Dummy Site ID
+    ///
+    let sampleSiteID: Int64 = 1234
+
+    /// Repeat always!
+    ///
+    override func setUp() {
+        network.removeAllSimulatedResponses()
+    }
+
+    func test_load_payment_gateways_return_all_gateways() throws {
+        // Given
+        let remote = PaymentGatewayRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "payment_gateways", filename: "payment-gateway-list")
+
+        // When
+        var gatewaysResult: Result<[PaymentGateway], Error>?
+        waitForExpectation { exp in
+            remote.loadAllPaymentGateways(siteID: sampleSiteID) { result in
+                gatewaysResult = result
+                exp.fulfill()
+            }
+        }
```

Comment:
nit: I've been using the `waitFor` more these days (thanks to @shiki's addition) to assign the future result to a let constant like in: https://github.com/woocommerce/woocommerce-ios/blob/b44413d7f7e369e4e1fb505164c03941bf73a775/WooCommerce/WooCommerceTests/ViewRelated/Products/Edit%20Product/Downloadable%20Files/ProductDownloadSettingsViewModelTests.swift#L48-L52

this also works, up to you!

Reply:
wow, neat!